pkg/pillar/zedUpload/ociutil/oci.go

```diff
@@ -81,14 +81,21 @@ func Manifest(registry, repo, username, apiKey string, client *http.Client, prgc
 }
 
 // PullBlob downloads a blob from a registry and save it as a file as-is.
-func PullBlob(registry, repo, hash, localFile, username, apiKey string, client *http.Client, prgchan NotifChan) (int64, error) {
+func PullBlob(registry, repo, hash, localFile, username, apiKey string, maxsize int64, client *http.Client, prgchan NotifChan) (int64, error) {
 	log.Infof("PullBlob(%s, %s, %s) to %s", registry, repo, hash, localFile)
 
 	var (
-		w io.Writer
-		r io.Reader
+		w        io.Writer
+		r        io.Reader
+		stats    UpdateStats
+		size     int64
+		finalErr error
 	)
 
+	// send out the maximum size as we understand it
+	stats.Size = maxsize
```

Comment:
Will we also have a way to enforce maxsize to that a bad OCI registry can't make us run out of disk space?

Reply:
We could. Right now it just does the following:

* accepts a maxsize from the config that was passed, and uses that as the maximum size for the root of the tree (manifest or index), and then the gets sizes from there
* if no size was given, uses size of 0, which means, accept whatever size the root manifest/index is, and then sizes from there

We need a better approach. I will move some of this conversation into the regular comments.